torch/distributed/_functional_collectives.py

```diff
@@ -244,6 +248,18 @@ def _expand_group(group: RANK_TYPES, tag: str = "") -> Tuple[str, List[int], int
 
     return (tag, rankset, group_size)
 
+def _are_we_tracing() -> bool:
+    mode = get_innermost_proxy_mode()
+    if mode is None:
+        return False
+    return mode.tracer is not None
+
+def _maybe_wrap_tensor(self):
```

Comment:
this should also be easy to support in dynamo:

we just have to implement 1 special case, for '_maybe_wrap_tensor' which always traces wait_tensor, and never traces the guts of _maybe_wrap_tensor or _are_we_tracing.